.github/workflows/deploy.yml

```diff
@@ -2,11 +2,18 @@ name: Deploy Client Application
 
 on:
   workflow_dispatch: # run when somebody pushes the button
+  push:
+    branches: # run when we merge to the "staging" branch
+      - staging
 jobs:
   build-and-deploy:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
+      # VERY IMPORTANT: if you run this on any branch other than master, it will
+      # deploy to staging
+    - name: Choose deployment environment
+      run: echo >> $GITHUB_ENV deploy_env=$( [[ "$GITHUB_REF" == "/refs/heads/main" ]] && echo "client" || echo "staging")
```

Comment:
what does "client" as a deployment env mean? Is that the same as test/dev/demo?

Reply:
It's the current deployed app name, which I didn't want to change for my own convenience. 🙂 My theory was that we use that as "stable" demo and the other as dev/test, but I'm totally happy to rename either or both. 

I can also make the workflow more self-documenting, as far as that goes--I'll take a whack this morning.